lib/mongodb.js

```diff
@@ -843,10 +843,22 @@ MongoDB.prototype.buildWhere = function(model, where) {
   return query;
 };
 
-MongoDB.prototype.buildSort = function(model, order) {
+MongoDB.prototype.buildSort = function(model, order, options) {
   var sort = {};
   var idName = this.idName(model);
-  if (!order) {
+
+  var modelClass = this._models[model];
+
+  var disableDefaultSort = this.settings.disableDefaultSort;
+  if (options && options.hasOwnProperty('disableDefaultSort')) {
+    disableDefaultSort = options.disableDefaultSort === true;
```

Comment:
@HugoPoi I found the code here is a little confusing. 
If I understand correctly, the precedence of the 3 configure flags is:

`options.disableDefaultSort` > `this.settings.disableDefaultSort` > `modelClass.settings.disableDefaultSort`

Shall we refactor the code to have a more clear logic? For example:

```js
// The flag `disableDefaultSort` might be set from 3 places
// Their precedence is: `options.disableDefaultSort` > `this.settings.disableDefaultSort` > `modelClass.settings.disableDefaultSort`

var disableDefaultSort = getDisableDefaultSort(
  options && options.disableDefaultSort,
  this.settings.disableDefaultSort,
  modelClass.settings.disableDefaultSort,
);

function getDisableDefaultSort(config1, config2, config3) {
  if (config1 != undefined)  return config1;
  if (config2 != undefined) return config2;
  return !!config3;
}
```